docs/source/configuration_file.md

```diff
@@ -0,0 +1,158 @@
+
+#### debugging
+Bool. Extended verbosity and intermediate outputs.
+
+## Loader parameters
+
+#### bids_path
+String. Path of the BIDS folder.
+
+#### target_suffix
+List. Suffix list of the derivative file containing the ground-truth of interest (e.g. [`"_seg-manual"`, `"_lesion-manual"`]). The length of this list controls the number of output channels of the model (i.e. `out_channel`). If the list has a length greater than 1, then a multi-class model will be trained.
+
+#### contrasts
+- `train_validation`: List. List of image contrasts (e.g. `T1w`, `T2w`) loaded for the training and validation. If `multichannel` is `true`, this list represents the different channels of the input tensors (i.e. its length equals model's `in_channel`). Otherwise, the contrasts are mixed and the model has only one input channel (i.e. model's `in_channel=1`).
+- `test`: List. List of image contrasts (e.g. `T1w`, `T2w`) loaded in the testing dataset. Same comment than for `train_validation` regarding `multichannel`.
+- `balance`: Dict. Enables to weight the importance of specific channels (or contrasts) in the dataset: e.g. `{"T1w": 0.1}` means that only 10% of the available `T1w` images will be included into the training/validation/test set. Please set `multichannel` to `false` if you are using this parameter.
+
+#### multichannel
+Bool. Indicated if more than a contrast (e.g. `T1w` and `T2w`) is used by the model. See details in both `train_validation` and `test` for the contrasts that are input.
+
+#### slice_axis
+Choice between `"sagittal"`, `"coronal"`, and `"axial"`.
+Sets the slice orientation for on which the model will be used.
+
+#### slice_filter
+Dict. Discard a slice from the dataset if it meets a condition, see below.
+- `filter_empty_input`: Bool. Discard slices where all voxel intensities are zeros.
+- `filter_empty_mask`: Bool. Discard slices where all voxel labels are zeros.
+
+#### roi
+Dict. of parameters about the region of interest
+- `suffix`: String. Suffix of the derivative file containing the ROI used to crop (e.g. `"_seg-manual"`) with `ROICrop` as transform. Please use `null` if you do not want to use an ROI to crop.
+- `slice_filter_roi`: int. If the ROI mask contains less than `slice_filter_roi` non-zero voxels, the slice will be discarded from the dataset. This feature helps with noisy labels, e.g., if a slice contains only 2-3 labeled voxels, we do not want to use these labels to crop the image. This parameter is only considered when using `"ROICrop"`.
+
+## Split dataset
+
+#### fname_split
+String. File name of the log (joblib) that contains the list of training/validation/testing subjects. This file can later be used to re-train a model using the same data splitting scheme.
+
+#### random_seed
+Int. Seed used by the random number generator to split the dataset between training/validation/testing. The use of the same seed ensure the same split between the sub-datasets, which is useful to reproduce results.
```

Comment:
```suggestion
Int. Seed used by the random number generator to split the dataset between training/validation/testing. The use of the same seed ensures the same split between the sub-datasets, which is useful to reproduce results.
```
the use of "Int." and "Integer" is inconsistent, but let's not spend too much additional time on this. At some point, it is possible that the json will be moved to a python class, with docstrings following strict convention for defining types, so no need to be "too" strict at this point